src/SymbolCollector.Core/FatBinaryReader.cs

```diff
@@ -0,0 +1,172 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace SymbolCollector.Core
+{
+    public class FatMachO : IDisposable
+    {
+        private readonly bool _deleteFilesOnDispose;
+
+        public FatMachO(bool deleteFilesOnDispose = false) => _deleteFilesOnDispose = deleteFilesOnDispose;
+
+        public FatHeader Header { get; set; }
+        public IEnumerable<string> MachOFiles { get; set; } = Enumerable.Empty<string>();
+
+        internal IEnumerable<string> FilesToDelete = Enumerable.Empty<string>();
+
+        public void Dispose()
+        {
+            if (_deleteFilesOnDispose)
+            {
+                foreach (var file in FilesToDelete)
+                {
+                    File.Delete(file);
+                }
+            }
+        }
+    }
+
+    // AKA multi-arch file/universal binary
+    public class FatBinaryReader
+    {
+        private readonly ILogger<FatBinaryReader> _logger;
+        public const uint FatObjectMagic = 0x_cafe_babe;
```

Comment:
Beware that this magic is endianness dependent and also exists with all bytes inverted (0xbebafeca). However, iirc, the header is always to be read as big endian. You’d have to play around with this a bit.

Reply:
I have them layed out in big endian. The header is indeed big endian.

The code already makes sure what I read from the object is represented in big endian before doing the comparison:

https://github.com/getsentry/symbol-collector/pull/5/files#diff-d8ffc31abc1022481a9723cf51db450bR120-R123

Reply:
<img width="764" alt="image" src="https://user-images.githubusercontent.com/1633368/70566558-26753700-1b49-11ea-8fdd-f752ffde51bb.png">